### PR TITLE
Update qrcode-generator extension

### PR DESCRIPTION
**Pull request by theOpheliart**
## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/qrcode-generator/src/index.tsx [hunk 1]
```diff
@@ -1,17 +1,18 @@
-import { Action, ActionPanel, Form, getPreferenceValues, open, showToast, Toast } from "@raycast/api";
+import { Action, ActionPanel, Form, getPreferenceValues, showToast, Toast } from "@raycast/api";
 import QRCode from "qrcode";
 import { useState } from "react";
 import { generateQRCode, getQRCodePath, QRCodeView } from "./utils";
 import { FormValidation, useForm } from "@raycast/utils";
-
+import { randomBytes } from "crypto";
```
Comment (darmiel) [inline extensions/qrcode-generator/src/index.tsx:6]: this import isn't really used anywhere

File: extensions/qrcode-generator/src/index.tsx [hunk 2]
```diff
@@ -1,17 +1,18 @@
...
+var open = require('mac-open');
```
Comment (darmiel) [inline extensions/qrcode-generator/src/index.tsx:7]:
```suggestion
import open from "mac-open";
```

(can be remove if using `showInFinder`)

File: extensions/qrcode-generator/package.json [hunk 3]
```diff
@@ -46,6 +69,8 @@
   "dependencies": {
     "@raycast/api": "^1.77.0",
     "@raycast/utils": "^1.16.0",
+    "mac-open": "^0.1.3",
+    "open": "^10.1.0",
```
Comment (darmiel) [inline extensions/qrcode-generator/package.json:73]: This dependency isn't used

File: extensions/qrcode-generator/src/utils.tsx [hunk 4]
```diff
@@ -31,11 +31,12 @@ export function QRCodeView({ qrData }: { qrData: string }) {
...
 export const getQRCodePath = (qrcodeUrl: string) => {
+  const fileType = getPreferenceValues().fileType;
   // `https://www.example.com/foo?bar=foo` -> `www.example.com`
   const filename = String(qrcodeUrl.match(/^(?:https?:\/\/)?(?:[^@/\n]+@)?(?:www\.)?([^:/\n]+)/gm)).replace(
     /^(?:https?:\/\/)?/gm,
     ""
   );
 
-  return `${homedir()}/Downloads/qrcode-${filename}.png`;
+  return `${homedir()}/Downloads/qrcode-${filename}.`+fileType;
```
Comment (darmiel) [inline extensions/qrcode-generator/src/utils.tsx:41]:
```suggestion
  return `${homedir()}/Downloads/qrcode-${filename}.${fileType}`;
```

File: extensions/qrcode-generator/package.json [hunk 5]
```diff
@@ -46,6 +69,8 @@
   "dependencies": {
     "@raycast/api": "^1.77.0",
     "@raycast/utils": "^1.16.0",
+    "mac-open": "^0.1.3",
```
Comment (darmiel) [inline extensions/qrcode-generator/package.json:72]: This dependency was last updated [10 years ago](https://github.com/weareinteractive/node-mac-open) and is unmaintained.

You can probably use Raycast's `showInFinder` utility function:
https://developers.raycast.com/api-reference/utilities#showinfinder

```diff
- open(path);
+ showInFinder(path);
```

File: extensions/qrcode-generator/src/index.tsx [hunk 6]
```diff
@@ -1,17 +1,18 @@
...
 interface FormValues {
   url: string;
   inline: boolean;
 }
 
 export default function Command() {
-  const [qrData, setQrData] = useState<string>();
-  const { primaryAction } = getPreferenceValues<Preferences.Index>();
+  const [qrData, setQrData] = useState<string>();randomBytes
+  const primAction= getPreferenceValues().primaryAction;
```
Comment (darmiel) [inline extensions/qrcode-generator/src/index.tsx:15]: please run formatter over the files

File: extensions/qrcode-generator/src/index.tsx [hunk 7]
```diff
@@ -23,8 +24,8 @@ export default function Command() {
         const path = getQRCodePath(values.url);
         QRCode.toFile(path, values.url)
           .then(() => {
-            showToast(Toast.Style.Success, "QRCode saved", `You can find it here: ${path}`);
-            open(path);
+            showToast(Toast.Style.Success, "Code saved", `You can find it here: ${path}`);
+            open(path, {R: true});
```
Comment (darmiel) [inline extensions/qrcode-generator/src/index.tsx:28]:
```suggestion
            showInFinder(path);
```

**Review by darmiel — CHANGES_REQUESTED** [review]

**Comment by pernielsentikaer** [comment]
What the status of this PR @theOpheliart

**Pull request closed without merging**